File: CloudflareFileStorage.cpp
#include <iostream>
#include <fstream>
#include <sstream>
#include <curl/curl.h>

class CloudflareFileStorage {
private:
    std::string workerUrl;

public:
    CloudflareFileStorage(const std::string &url) {
        workerUrl = url.back() == '/' ? url : url + "/";
    }

    std::string uploadFile(const std::string &filePath) {
        std::ifstream file(filePath, std::ios::binary);
        if (!file) {
            throw std::runtime_error("File not found: " + filePath);
        }

        CURL *curl = curl_easy_init();
        if (!curl) {
            throw std::runtime_error("CURL initialization failed");
        }

        curl_mime *mime = curl_mime_init(curl);
        curl_mimepart *part = curl_mime_addpart(mime);
        curl_mime_filedata(part, filePath.c_str());
        curl_mime_type(part, "application/octet-stream");

        curl_easy_setopt(curl, CURLOPT_URL, (workerUrl + "upload").c_str());
        curl_easy_setopt(curl, CURLOPT_MIMEPOST, mime);

        CURLcode res = curl_easy_perform(curl);
        curl_mime_free(mime);
        curl_easy_cleanup(curl);

        if (res != CURLE_OK) {
            throw std::runtime_error("CURL request failed");
        }

        return "File uploaded successfully";
    }

    std::string downloadFile(const std::string &fileName, const std::string &savePath) {
        CURL *curl = curl_easy_init();
        if (!curl) {
            throw std::runtime_error("CURL initialization failed");
        }

        curl_easy_setopt(curl, CURLOPT_URL, (workerUrl + "download/" + fileName).c_str());

        std::ofstream outFile(savePath + "/" + fileName, std::ios::binary);
        curl_easy_setopt(curl, CURLOPT_WRITEDATA, &outFile);

        CURLcode res = curl_easy_perform(curl);
        curl_easy_cleanup(curl);

        if (res != CURLE_OK) {
            throw std::runtime_error("CURL request failed");
        }

        return "File downloaded successfully";
    }

    std::string deleteFile(const std::string &fileName) {
        CURL *curl = curl_easy_init();
        if (!curl) {
            throw std::runtime_error("CURL initialization failed");
        }

        curl_easy_setopt(curl, CURLOPT_URL, (workerUrl + "delete/" + fileName).c_str());
        curl_easy_setopt(curl, CURLOPT_CUSTOMREQUEST, "DELETE");

        CURLcode res = curl_easy_perform(curl);
        curl_easy_cleanup(curl);

        if (res != CURLE_OK) {
            throw std::runtime_error("CURL request failed");
        }

        return "File deleted successfully";
    }
};

File: CloudflareFileStorage.java
import java.io.*;
import java.net.*;
import java.nio.file.*;
import java.util.*;

public class CloudflareFileStorage {
    private final String workerUrl;

    public CloudflareFileStorage(String workerUrl) {
        this.workerUrl = workerUrl.endsWith("/") ? workerUrl : workerUrl + "/";
    }

    public String uploadFile(String filePath) throws IOException {
        File file = new File(filePath);
        if (!file.exists()) {
            throw new FileNotFoundException("File not found: " + filePath);
        }

        URL url = new URL(workerUrl + "upload");
        HttpURLConnection connection = (HttpURLConnection) url.openConnection();
        connection.setDoOutput(true);
        connection.setRequestMethod("POST");
        connection.setRequestProperty("Content-Type", "application/octet-stream");

        try (OutputStream os = connection.getOutputStream(); FileInputStream fis = new FileInputStream(file)) {
            byte[] buffer = new byte[1024];
            int bytesRead;
            while ((bytesRead = fis.read(buffer)) != -1) {
                os.write(buffer, 0, bytesRead);
            }
        }

        try (BufferedReader in = new BufferedReader(new InputStreamReader(connection.getInputStream()))) {
            return in.readLine();
        }
    }

    public byte[] downloadFile(String fileName, String savePath) throws IOException {
        URL url = new URL(workerUrl + "download/" + fileName);
        HttpURLConnection connection = (HttpURLConnection) url.openConnection();
        connection.setRequestMethod("GET");

        try (InputStream in = connection.getInputStream()) {
            Path path = Paths.get(savePath, fileName);
            Files.copy(in, path, StandardCopyOption.REPLACE_EXISTING);
            return Files.readAllBytes(path);
        }
    }

    public String deleteFile(String fileName) throws IOException {
        URL url = new URL(workerUrl + "delete/" + fileName);
        HttpURLConnection connection = (HttpURLConnection) url.openConnection();
        connection.setRequestMethod("DELETE");

        try (BufferedReader in = new BufferedReader(new InputStreamReader(connection.getInputStream()))) {
            return in.readLine();
        }
    }
}
